src/components/RadioList.tsx

```diff
@@ -6,26 +6,58 @@ interface RadioListProps {
   children: ReactNode;
 }
 
+interface RadioListConditionalProps {
+  className?: string;
+  children: ReactNode;
+}
+
 interface RadioListItemProps {
   id: string;
   name: string;
   value: string;
-  text: string;
+  children: ReactNode;
+  inputHtmlAttributes?: Record<string, string>;
```

Comment:
Required for setting custom `data-*` HTML5 attributes that the conditional radio button requires.

See the [docs](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) for further information.